### ISSUE TITLE
datetime-local, date, month, time <input>s are taller than other <input>s in Chrome/Edge

### ISSUE THREAD

**Issue by heruan** [twbs/bootstrap#17308]
Why inside an `input-group` the `input` type `datetime-local` gets a different height from other `form-control`s? See https://jsfiddle.net/02z6gh62/2/ (I'm using Chrome)

I've found this
````css
@media screen and (-webkit-min-device-pixel-ratio: 0)
.input-group-sm input[type=date].form-control,
.input-group-sm input[type=time].form-control,
.input-group-sm input[type=datetime-local].form-control,
.input-group-sm input[type=month].form-control,
input[type=date].input-sm, input[type=time].input-sm,
input[type=datetime-local].input-sm,
input[type=month].input-sm {
    line-height: 1.95rem;
}
````
and setting `line-height` to `1rem` works perfectly fine in Chrome, then I wonder why it's forced to 1.95.

**Comment by twbs-lmvtfy** [comment]
Hi @heruan!

You appear to have posted a live example (https://fiddle.jshell.net/02z6gh62/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 46, column 17 thru column 76: Duplicate ID ```` from ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by heruan** [comment]
Revised example https://jsfiddle.net/02z6gh62/1/

**Comment by twbs-lmvtfy** [comment]
Hi @heruan!

You appear to have posted a live example (https://fiddle.jshell.net/02z6gh62/1/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 40, column 17: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by hnrch02** [comment]
We do this for iOS Safari ([see this comment](https://github.com/twbs/bootstrap/blob/210e2b8100e0a90e097cd98f1abd7c3bfdc72a8b/scss/_forms.scss#L88-L114)) though that hack seems to also include other WebKit-based browsers...

**Comment by cvrebert** [comment]
"The Safari 7.1+ Hack" from http://browserstrangeness.bitbucket.org/css_hacks.html#safari seems to work as advertised and can be simplified slightly to:
```css
_::-webkit-full-page-media, .actual_selector_here {
    /* Safari-only CSS here */
}
```
So, we could try using that.

**Comment by patrickhlauke** [comment]
Worth noting (here?) that in Edge (using the testcases from https://github.com/twbs/bootstrap/pull/18784)  `datetime-local`, `date`, `month` and `time` (but not, strangely, `week`) display taller than other inputs.

> ![capture](https://cloud.githubusercontent.com/assets/895831/12169328/aea567be-b52f-11e5-98b0-00544a70089c.PNG)

**Comment by patrickhlauke** [comment]
and ditto for chrome/windows (`datetime-local`, `date`, `month` and `time` but not `week`) 

> ![capture](https://cloud.githubusercontent.com/assets/895831/12169418/32eb400c-b530-11e5-872c-41b99472de79.PNG)

**Comment by cvrebert** [comment]
I conducted more testing (see http://browserbu.gs/css-hacks/webkit-full-page-media/ ) and the `_::-webkit-full-page-media` hack for targeting only Safari is very solid, so I'm gonna make a PR for that.

**Comment by cvrebert** [comment]
Fixed by #18816.
